From: Koala Yeung <koalay@gmail.com>
Date: Wed, 22 Apr 2020 21:52:18 +0800
Subject: Fix language fallback for Chinese (Hong Kong)
Origin: https://github.com/micahflee/torbrowser-launcher/pull/464

* Add a way to manually override the language detection
  for some of the languages.
* Add manual language overriding for zh-HK (to zh-TW).

(cherry picked from commit e92e11e6d7cde178d4aad3635fe84a32e5a32069)
---
 torbrowser_launcher/common.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/torbrowser_launcher/common.py b/torbrowser_launcher/common.py
index 1679a96..b141894 100644
--- a/torbrowser_launcher/common.py
+++ b/torbrowser_launcher/common.py
@@ -74,12 +74,20 @@ class Common(object):
 
         # figure out the language
         available_languages = ['ar', 'ca', 'da', 'de', 'en-US', 'es-ES', 'fa', 'fr', 'ga-IE', 'he', 'id', 'is', 'it', 'ja', 'ko', 'nb-NO', 'nl', 'pl', 'pt-BR', 'ru', 'sv-SE', 'tr', 'vi', 'zh-CN', 'zh-TW']
+
+        # a list of manually configured language fallback overriding
+        language_overrides = {
+            'zh-HK': 'zh-TW',
+        }
+
         locale.setlocale(locale.LC_MESSAGES, '')
         default_locale = locale.getlocale(locale.LC_MESSAGES)[0]
         if default_locale is None:
             self.language = 'en-US'
         else:
             self.language = default_locale.replace('_', '-')
+            if self.language in language_overrides:
+                self.language = language_overrides[self.language]
             if self.language not in available_languages:
                 self.language = self.language.split('-')[0]
                 if self.language not in available_languages:
